nixos/modules/services/mail/postfix.nix

```diff
@@ -127,11 +127,14 @@ let
     #               (yes)   (yes)   (no)    (never) (100)
     # ==========================================================================
     smtp      inet  n       -       n       -       -       smtpd
-    #submission inet n       -       n       -       -       smtpd
-    #  -o smtpd_tls_security_level=encrypt
-    #  -o smtpd_sasl_auth_enable=yes
-    #  -o smtpd_client_restrictions=permit_sasl_authenticated,reject
-    #  -o milter_macro_daemon_name=ORIGINATING
+  '' + optionalString cfg.enableSubmission ''
+    submission inet n       -       n       -       -       smtpd
+      -o smtpd_tls_security_level=encrypt
+      -o smtpd_sasl_auth_enable=yes
+      -o smtpd_client_restrictions=permit_sasl_authenticated,reject
+      ${cfg.extraSubmissionOptions}
+  ''
+  + ''
```

Comment:
IMHO, this solution hard-codes way too many settings to be useful in a general case. For example, how would I use that if I don't want TLS encryption? Boiling all this complexity of local mail injection down to a single on/off switch seems like it's limiting the applicability of the setting way too much.